### PR TITLE
feat: add Carbon Footprint Predictor using sklearn pipeline

### LINKED ISSUE #1637: Feat: Add an AI/ML Carbon Footprint Predictor but integrate it with the web only
I would like to implement a modular Machine Learning pipeline using Python and `scikit-learn` that predicts an individual's annual Carbon Footprint (CO2 emission) based on daily lifestyle factors.

The implementation will feature:
1. **Handling Categorical Data:** Implement One-Hot Encoding for non-numeric variables (e.g., Diet Type: Vegan/Meat-Eater, Primary Transport: Electric/Petrol/Public).
2. **Feature Scaling:** Apply standard scaling to numeric attributes (e.g., Monthly Electricity Bill, Annual Flight Hours).
3. **Regression Pipeline:** Train and compare baseline models like `LinearRegression` or a simple `RandomForestRegressor`.
4. **Evaluation Metrics:** Evaluate performance using Mean Absolute Error (MAE) and $R^2$ Score to teach contributors how to read regression performance metrics.

As Alternatives
- **Classification Approach:** Grouping footprints into "Low/Medium/High" categories. However, keeping it as a regression task (predicting the exact metric tons of CO2) makes it a far better tool for teaching continuous data modeling.

As a beginner passionate about learning AI/ML workflows, I would absolutely love to work on this issue! It matches my current learning path perfectly, and I am committed to writing clean, well-documented code that aligns with the project's quality standards. Please assign this issue to me under GSSoC 2026.

### PR DESCRIPTION
**Pull request by Sai-Ramya-prog**
<html>
<body>
<p class="font-claude-response-body break-words whitespace-normal">Closes #1637</p>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Title:</strong> feat: Add Carbon Footprint Predictor using scikit-learn Pipeline</p>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Description:</strong><br>
Added a new ML project — Carbon Footprint Predictor — that predicts an individual's annual CO₂ emissions based on daily lifestyle factors.</p>
<p class="font-claude-response-body break-words whitespace-normal"><strong>What's included:</strong></p>
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="font-claude-response-body whitespace-normal break-words pl-2">Modular scikit-learn Pipeline with ColumnTransformer</li>
<li class="font-claude-response-body whitespace-normal break-words pl-2">OneHotEncoding for categorical features (Diet, Transport, etc.)</li>
<li class="font-claude-response-body whitespace-normal break-words pl-2">StandardScaler for numerical features</li>
<li class="font-claude-response-body whitespace-normal break-words pl-2">Two models trained and compared:
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="font-claude-response-body whitespace-normal break-words pl-2">Linear Regression (baseline)</li>
<li class="font-claude-response-body whitespace-normal break-words pl-2">Random Forest Regressor</li>
</ul>
</li>
<li class="font-claude-response-body whitespace-normal break-words pl-2">Evaluation using MAE and R² Score</li>
<li class="font-claude-response-body whitespace-normal break-words pl-2">Visualizations: distribution plot, correlation heatmap, actual vs predicted</li>
</ul>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Results:</strong></p>
<div class="overflow-x-auto w-full px-2 mb-6">
Model | MAE | R²
-- | -- | --
Linear Regression | 285.19 | 0.89
Random Forest | 304.29 | 0.90

</div>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Dataset:</strong> <a class="underline underline underline-offset-2 decoration-1 decoration-current/40 hover:decoration-current focus:decoration-current" href="https://www.kaggle.com/datasets/dumanmesut/individual-carbon-footprint-calculation">Kaggle — Individual Carbon Footprint Calculation</a></p>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Files Added:</strong></p>
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="font-claude-response-body whitespace-normal break-words pl-2"><code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">Carbon_Footprint_Predictor/carbon_footprint_predictor.ipynb</code></li>
<li class="font-claude-response-body whitespace-normal break-words pl-2"><code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">Carbon_Footprint_Predictor/README.md</code></li>
</ul>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Checklist:</strong></p>
<ul class="contains-task-list">
<li class="task-list-item"><input disabled="" type="checkbox" checked=""> Notebook runs without errors</li>
<li class="task-list-item"><input disabled="" type="checkbox" checked=""> README.md included</li>
<li class="task-list-item"><input disabled="" type="checkbox" checked=""> Dataset source credited (Kaggle)</li>
<li class="task-list-item"><input disabled="" type="checkbox" checked=""> Follows project folder structure</li>
<li class="task-list-item"><input disabled="" type="checkbox" checked=""> GSSoC 2026 contribution</li>
<li class="task-list-item"><input disabled="" type="checkbox" checked=""> Related to valid issue #1637</li>
</ul>
<p class="font-claude-response-body break-words whitespace-normal"><strong>Participant Info:</strong></p>
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="font-claude-response-body whitespace-normal break-words pl-2">GitHub username: Sai-Ramya-prog</li></ul>
</body>

### CODE REVIEW

**Comment by Sibam-Paul** [comment]
I think we don't need this Issue it's a common one

**Comment by Sai-Ramya-prog** [comment]
Can you please suggest me any changes ? I try to update this

**Comment by Sibam-Paul** [comment]
> Can you please suggest me any changes ? I try to update this

add a feature or model which is not there or not common one 
simple projects we are not accepting as there are many simple models present already @Sai-Ramya-prog

**Comment by Sai-Ramya-prog** [comment]
Then what about this pr should I leave it

**Comment by Sibam-Paul** [comment]
> Then what about this pr should I leave it

Wait i will inform @Sai-Ramya-prog join the discussion tab in discord please of this Project

**Comment by Niketkumardheeryan** [comment]
Hi @Sai-Ramya-prog, you can integrate it with the web, or you can pick any other issue

**Comment by Sibam-Paul** [comment]
@Sai-Ramya-prog please integrate into web as told in the above comment 
then raise the PR again I am closing this @Sai-Ramya-prog
